pkg/provisioner/ironic/ironic.go

```diff
@@ -1368,3 +1371,107 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 
 	return result, nil
 }
+
+// Describes the current availability status of a generic dependency
+// required by the provisioner to operate correctly
+type provisionerDependencyStatus struct {
+	// The name of the dependency
+	name string
+	// If ready, the dependency is available to process incoming requests
+	ready bool
+	// Can provide additional info if the dependency is not ready
+	err error
+}
+
+// Check to verify dependency status
+type provisionerDependencyCheck func(chan<- provisionerDependencyStatus)
+
+// IsReady checks if the provisioning backend is available
+func (p *ironicProvisioner) IsReady() (bool, error) {
+	p.log.Info("verifying ironic provisioner dependencies")
+
+	checks := []provisionerDependencyCheck{
+		p.checkIronic,
+		p.checkInspector,
+	}
+
+	var done = make(chan provisionerDependencyStatus, len(checks))
+	defer close(done)
+
+	for _, check := range checks {
+		go check(done)
```

Comment:
Mostly to reduce as much as possible the `IsReady` execution time to minimize its impact on the reconcile loop. The underlying services were also independent, so there was also no strict reasons to perform all the checks sequentially. That approach could also be somehow useful when/if adding new checks in future, so that the overall execution time will not greater than the slower check in the list.

Reply:
OK. I'm more worried about clarity than performance, right now. We get a separate goroutine per host being reconciled, and none of the work takes very long.